src/unitxt/operators.py

```diff
@@ -213,6 +213,97 @@ def process(
         return flatten_dict(instance, parent_key=self.parent_key, sep=self.sep)
 
 
+class WholeInputFormatter(StreamInstanceOperator):
+    r"""Generates the whole input to the model, from constant strings that are given as args, and from values found in specified fields of the instance.
+
+    WholeInputFormatter expects its input instance to have been processed by a Template StreamInstanceOperator
+    (from templates.py), or any extension thereof, and hence to contain a field named "source", and to have
+    been processed by RenderInstruction StreamInstanceOperator (from renderer.py) and hence to contain a non None
+    field "instruction".
+    Demos are assumed to be included as well, in a field whose name is specified by argument 'demos_field'. That
+    field is assumed to contain a list of dicts, each containing keys: "source" and "target" of a single demo.
+    And finally, system_prompt is also assumed to be included
+    in the instance, in a field whose name is specified by argument 'system_prompt'.
+    The value in field "source" of the input instance describes (verbalizes) original values in the instance (as read
+    from the source dataset), in the context of the underlying task.
+    WholeInputFormatter makes this (verbalization of the) input into a whole-input-to-the-model, by combining it
+    with instruction, demos, and system_prompt, read from the input instance, through both formatting args
+    formatting arguments.
+
+    Args:
+        to_field (str): the name of the field into which the formatted model's input is to be stored
+        demos_field (str): the name of the field that contains the demos, being dicts with "source" and "target" keys
+        system_prompt (str): the name of the field containing an introductory text and or xml tags
+        demo_format (str): formatting string for a single demo, combining fields source and target
+        general_format (str) overall output format, combinig system_prompt, instruction, demos, and source
+    """
+
+    to_field: str
+    demos_field: str = None
+    system_prompt: str = None
+    demo_format: str = (
+        "{source}\n{target}\n\n"  #  example: "User: {source}\nAgent: {target}\n\n"
+    )
+    general_format: str = (
+        "<SYS>{system_prompt}</SYS>\n{instruction}\n{demos}\n{source}\n"
+    )
+    instruction_prefix: str = ""  # potential prepend for non "" instruction. necessary for backward compatibility with iclformat
```

Comment:
Not needed, in my view.

Reply:
I had a problem with ICLFormat, with which I wanted to keep backward compatibility.
ICLFormat does prepend arg `instruction_prefix` (, e.g. "Instruction: ") only when: (1) the instruction comes before the demos, not after, **and** (2) the instruction is not "".   Now, if I move this "Instruction: " to be part of `general_format`,   "Instrucion: " will show also when an instance does not have instruction, namely: in cases where instruction is "".

Reply:
I see.  Thanks for clarifying.  I think today "Instruction" is really used for system prompt, and in this case, there is now need to add "Instruction:".  It can be explicitly done.